src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,49 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    private MachineProducer<? extends Machine> machineProducer;
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type.equals(Bulldozer.class)) {
+            machineProducer = new BulldozerProducer();
+            List<? extends Machine> machines = machineProducer.get();
```

Comment:
Any sense of parametrization here
```suggestion
            machineProducer = new BulldozerProducer();
            List<Bulldozer> machines = machineProducer.get();
```